test/integration/targets/iam_user/tasks/main.yml

```diff
@@ -215,14 +397,69 @@
       that:
         - iam_user_info.iam_users | length == 0
 
+  - name: remove group
+    iam_group:
+      name: '{{ test_group }}'
+      state: absent
+    register: iam_group
+
+  - name: assert that group was removed
+    assert:
+      that:
+        - iam_group.changed
+        - iam_group
+
+  - name: Test remove group again (idempotency)
+    iam_user:
+      name: "{{ test_group }}"
+      state: absent
+    register: iam_group
+
+  - name: assert that group remove is not changed
+    assert:
+      that:
+        - not iam_group.changed
+
+  - name: Remove user with attached policy
+    iam_user:
+      name: "{{ test_user }}"
+      state: absent
+    register: iam_user
+
+  - name: get info on IAM user(s) after deleting
+    iam_user_info:
+      group: '{{ test_user }}'
+    ignore_errors: yes
+    register: iam_user_info
+
+  - name: Assert user was removed
+    assert:
+      that:
+        - iam_user.changed
+        - "'cannot be found' in iam_user_info.msg"
+
+  - name: Remove user with attached policy (idempotent)
+    iam_user:
+      name: "{{ test_user }}"
+      state: absent
+    ignore_errors: yes
+    register: iam_user
+
+  - name: Assert user was removed
+    assert:
+      that:
+        - not iam_user.changed
+
   always:
   - name: remove group
     iam_group:
       name: '{{ test_group }}'
       state: absent
+    ignore_errors: yes
 
```

Comment:
Because removing users with policies has been buggy, we should probably remove the managed policies before removing the user, otherwise regressions / testing the old buggy version will leave things behind

Reply:
This is an `iam_group` task, is that where you intended to comment? The iam_user module should be fixed for removing users with policies by f0cadb9843dc2103e7f223052812eed993257653 and we definitely want tests for any of those new code paths possible. And it's better to have thorough tests and make the terminator work a little bit and know something is wrong if there's a bug rather than structuring our tests to cooperate with buggy code.

Reply:
Additionally, afaict we don't actually have any module that can detach a managed policy as a stand-alone task.  I'd have to do:
```
iam_user:
  name: '{{ item }}'
  state: present
  purge_policy: yes
with_items: "{{ test_users }}"
```
If the user removal test currently at ln423 has not failed, this is going to re-create the user.